flink-runtime/src/test/java/org/apache/flink/runtime/operators/ReduceTaskTest.java

```diff
@@ -50,7 +50,7 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.assertj.core.api.Assertions.fail;
 
-public class ReduceTaskTest extends DriverTestBase<RichGroupReduceFunction<Record, Record>> {
+public class ReduceTaskTest extends DriverTestBase<GroupReduceFunction<Record, Record>> {
```

Comment:
Same here: Were those changes made accidentally?

Reply:
This is necessary because we have change the parameter from `Class stubClass` to `Class<? extends S> stubClass`.

The `ReduceTaskTest`'s `MockCombiningReduceStub` implements `GroupReduceFunction`.
If we continue to use `DriverTestBase<RichGroupReduceFunction<Record, Record>>`, we will exceed the specified boundaries.
https://github.com/apache/flink/blob/c61c09e464073fae430cab2dd56bd608f9d275fd/flink-runtime/src/test/java/org/apache/flink/runtime/operators/ReduceTaskTest.java#L316-L318

Reply:
Ah, I see. I missed that one :+1: Even though, one could argue that instead of fixing the generic you could just switch to the not-deprecated `DriverTestBase#testDriver(Driver, S)` method which you introduced. But I leave it up to you whether you still want to change that. :shrug:

Reply:
Thanks @XComp for your continuous help. 

Using the newly introduced method `testDriver(Driver, S)` still requires fixing the generics because `S` also restricts the scope of the class. The old method parameter `testDriver(Driver driver, Class stubClass)` does not have any restrictions on the class, so it can be compiled successfully, but it may increase the possibility of errors occurring.